rs/ic_os/config/src/types.rs

```diff
@@ -120,3 +114,37 @@ pub struct Logging {
     /// Space-separated list of tags to apply to exported log records.
     pub elasticsearch_tags: Option<String>,
 }
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub struct NetworkSettings {
+    pub ipv6_config: Ipv6Config,
+    pub ipv4_config: Option<Ipv4Config>,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub struct Ipv4Config {
+    pub address: Ipv4Addr,
+    pub gateway: Ipv4Addr,
+    pub prefix_length: u8,
+    pub domain: String,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub enum Ipv6Config {
```

Comment:
The test_serialize_and_deserialize unit test constructs the whole config object and serializes/deserializes it. Do you think this is sufficient?

Reply:
I think we're OK on coverage from those tests, it's just that it's hard to tell what the serialized value looks like.

Maybe this will be helpful as we think about backwards compatibility? Putting it into a test is just an easy way to see it from the code, but we'll have to keep it up to date. Maybe it is easy enough just to run the tool directly instead